### PR TITLE
Implemented thread_join and thread_reap

### PR DESCRIPTION
**Pull request by rafalcieslak**
See #216.

### CODE REVIEW

**Comment by rafalcieslak** [comment]
The test results indicate that the idle thread is incorrectly marked as `RUNNING`. I've fixed this issue in #224.

File: sys/thread.c [hunk 1]
```diff
@@ -14,11 +14,17 @@ typedef TAILQ_HEAD(, thread) thread_list_t;
 /* TODO: Synchronize access to the list */
 static thread_list_t all_threads;
 
+mtx_t zombie_threads_mtx;
```
Comment (cahirwpz) [inline sys/thread.c]: Shouldn't it be defined static?

File: sys/thread.c [hunk 2]
```diff
@@ -14,11 +14,17 @@ typedef TAILQ_HEAD(, thread) thread_list_t;
 /* TODO: Synchronize access to the list */
```
Comment (cahirwpz) [inline sys/thread.c]: Could you fix that in this PR as well?

File: tests/thread_exit.c [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#include <stdc.h>
+#include <clock.h>
+#include <thread.h>
+#include <sched.h>
+#include <ktest.h>
+
+int exit_time[] = {100, 200, 150};
```
Comment (cahirwpz) [inline tests/thread_exit.c]: All (?) global variables in test modules should be defined static to avoid symbol clashes.

**Pull request merged**